runelite-api/src/main/java/net/runelite/api/coords/WorldPoint.java

```diff
@@ -261,6 +261,40 @@ public int distanceTo2D(WorldPoint other)
 		return Math.max(Math.abs(getX() - other.getX()), Math.abs(getY() - other.getY()));
 	}
 
+	/**
+	 * Gets the straight-line distance between this point and another.
+	 * <p>
+	 * If the other point is not on the same plane, this method will return
+	 * {@link Float#MAX_VALUE}. If ignoring the plane is wanted, use the
+	 * {@link #distanceTo2DHypotenuse(WorldPoint)} method.
+	 *
+	 * @param other other point
+	 * @return the straight-line distance
+	 */
+	public float distanceToHypotenuse(WorldPoint other)
+	{
+		if (other.plane != plane)
+		{
+			return Float.MAX_VALUE;
+		}
+
+		return distanceTo2DHypotenuse(other);
+	}
+
+	/**
+	 * Find the straight-line distance from this point to another point.
+	 * <p>
+	 * This method disregards the plane value of the two tiles and returns
+	 * the simple distance between the X-Z coordinate pairs.
+	 *
+	 * @param other other point
+	 * @return the straight-line distance
+	 */
+	public float distanceTo2DHypotenuse(WorldPoint other)
```

Comment:
LocalPoint already contains distanceTo, so you should probably use that

Reply:
I couldn't get LocalPoint to work (without tweaking the fromScene method in LocalPoint) after trying for quite a while so I've reverted to using WorldPoint and it works great. I need that method to calculate a proper diagonal distance and detect if the player is one tile away to the side.

If you're okay with that then I think I'm done, unless you'd like any other changes.